kernel/aster-nix/src/syscall/mount.rs

```diff
@@ -0,0 +1,163 @@
+// SPDX-License-Identifier: MPL-2.0
+
+use super::SyscallReturn;
+use crate::{
+    fs::{
+        fs_resolver::{FsPath, AT_FDCWD},
+        path::Dentry,
+        utils::InodeType,
+    },
+    prelude::*,
+    syscall::constants::MAX_FILENAME_LEN,
+    util::read_cstring_from_user,
+};
+
+pub fn sys_mount(
+    devname_addr: Vaddr,
+    dirname_addr: Vaddr,
+    fstype_addr: Vaddr,
+    flags: u64,
+    data: Vaddr,
```

Comment:
Although our current implementation of the filesystem does not require `data`, I propose to comment it out here for subsequent refinement. 
It would be better to pass it to the `target_dentry.mount(fs)?` and let the file system that needs to use it judge for itself.

Reply:
Thank you for your thoughtful suggestion. I agree that in our current implementation, the data field is not required for handling the filesystem, so I've added a comment to clarify this. Regarding your proposal to pass `data` to `target_dentry.mount(fs)?`, I've considered the following points:

**Minimal Viable Feature Set**: For now, I am only focusing on implementing the dummy do_new_mount functionality, as `bind mount` and `move mount` do not require processing of `data`. The `data` field is not essential for this basic functionality, so to minimize code changes and maintain simplicity, I believe it is reasonable to defer handling `data` for the time being.

**Filesystem-Specific Handling**: The `data` is closely tied to the specifics of the filesystem in question. The `mount` method within the `impl Dentry` is primarily responsible for creating a `MountNode` struct and performing the mounting action. Even if we pass `data` to this method, it would be challenging to process it effectively without additional context from the filesystem implementation.